app/battle_preparation/knight.py

```diff
@@ -0,0 +1,24 @@
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.equipment = None
+        self.liquid = None
+        self.weapon = None
+
+    def get_ready_for_battle(self,
+                             armor_parts: list,
+                             miracle_potion: dict,
+                             divine_weapon: dict) -> None:
+        from app.battle_preparation.armour import Armour
+        armour = Armour.prepare_armour(armor_parts)
+        armour.consider_armour_effect(self)
+        from app.battle_preparation.potion import Potion
+        potion = Potion.drink_potion(miracle_potion)
+        if potion:
+            potion.consider_potion_effect(self)
+        from app.battle_preparation.weapon import Weapon
+        weapon = Weapon.prepare_weapon(divine_weapon)
+        weapon.consider_weapon(self)
```

Comment:
I a bit disagree with this approach. You create class instances in order to just call `consider` method and for attributes self.protection, equipment, liquid, weapon you save only values(numbers and for potion it's potion name for some reason). So you actually don't need class instances, just some static methods. But from the other side you don't save class instances and it means if you leave battle function, you'll not be able to access what list of armours this knight has and will need to extract it again from config. I recommend to save this boosts as class instance / list of instances and have some methods/properties in Knight that will call methods of these instances(weapon, armour, etc) that will return boost(for example number of protection)
In case you want to get info about what armour this knight has, you'll be able to get it using knight.armour -> list[Armour] and calculate that boost whenever you want

Reply:
Just wanted to separate this modules in order to make this boosts more flexible